kernel/src/process_standard.rs

```diff
@@ -548,8 +548,13 @@ impl<C: Chip, D: 'static + ProcessStandardDebug> Process for ProcessStandard<'_,
     }
 
     fn ready(&self) -> bool {
-        self.tasks.map_or(false, |ring_buf| ring_buf.has_elements())
-            || self.state.get() == State::Running
+        self.state.get() == State::Running
+            || if let State::YieldedFor { ready: true, .. } = self.state.get() {
+                true
+            } else {
+                false
+            }
+            || self.tasks.map_or(false, |ring_buf| ring_buf.has_elements())
```

Comment:
Doesn't this last case need to ensure that state is NOT YieldedFor? Otherwise a state YieldedFor which is not ready could still have unrelated tasks in the ring buffer and return true.